src/storage/posix/storage.c

```diff
@@ -465,39 +388,46 @@ storagePosixPathCreate(
 }
 
 /**********************************************************************************************************************************/
-static bool
-storagePosixPathExists(THIS_VOID,  const String *path, StorageInterfacePathExistsParam param)
+typedef struct StoragePosixPathRemoveData
 {
-    THIS(StoragePosix);
+    StoragePosix *driver;                                           // Driver
+    const String *path;                                             // Path
+} StoragePosixPathRemoveData;
 
-    FUNCTION_LOG_BEGIN(logLevelTrace);
-        FUNCTION_LOG_PARAM(STORAGE_POSIX, this);
-        FUNCTION_LOG_PARAM(STRING, path);
-        (void)param;                                                // No parameters are used
-    FUNCTION_LOG_END();
+static void
+storagePosixPathRemoveCallback(void *callbackData, const StorageInfo *info)
+{
+    FUNCTION_TEST_BEGIN();
+        FUNCTION_TEST_PARAM_P(VOID, callbackData);
+        FUNCTION_TEST_PARAM_P(STORAGE_INFO, info);
+    FUNCTION_TEST_END();
 
-    ASSERT(this != NULL);
-    ASSERT(path != NULL);
+    ASSERT(callbackData != NULL);
+    ASSERT(info != NULL);
 
-    bool result = false;
+    if (!strEqZ(info->name, "."))
+    {
+        StoragePosixPathRemoveData *data = callbackData;
 
-    // Attempt to stat the file to determine if it exists
-    struct stat statPath;
+        String *file = strNewFmt("%s/%s", strPtr(data->path), strPtr(info->name));
 
-    // Any error other than entry not found should be reported
-    if (stat(strPtr(path), &statPath) == -1)
-    {
-        if (errno != ENOENT)
-            THROW_SYS_ERROR_FMT(PathOpenError, "unable to stat '%s'", strPtr(path));
+        // Rather than stat the file to discover what type it is, just try to unlink it and see what happens
+        if (unlink(strPtr(file)) == -1)
+        {
+            // These errors indicate that the entry is actually a path so we'll try to delete it that way
+            if (errno == EPERM || errno == EISDIR)              // {uncovered_branch - no EPERM on tested systems}
+            {
+                storageInterfacePathRemoveP(data->driver, strNewFmt("%s/%s", strPtr(data->path), strPtr(info->name)), true);
```

Comment:
`String *file` is already set to `strNewFmt("%s/%s", strPtr(data->path), strPtr(info->name))` so why can't you use file in this function call? 
`storageInterfacePathRemoveP(data->driver, file, true);`

Reply:
No reason that I can see. Fixed.